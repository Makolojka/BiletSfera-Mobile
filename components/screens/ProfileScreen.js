import React, { useEffect, useState } from 'react';
import {Text, View, StyleSheet, Image} from 'react-native';
import jwtDecode from 'jwt-decode'; // Import jwt-decode library
import AsyncStorage from '@react-native-async-storage/async-storage';
import AuthService from "../../services/AuthService";
import {COLORS} from "./Colors";
import MaterialIcons from "react-native-vector-icons/MaterialCommunityIcons"; // Import AsyncStorage

const ProfileScreen = () => {
    const [userDetails, setUserDetails] = useState({});

    useEffect(() => {
        async function fetchUserDetails() {
            try {
                const userDetailsFromToken = await AuthService.getUserDetails();
                if(userDetailsFromToken)
                {
                    setUserDetails(userDetailsFromToken);
                }
            } catch (error) {
                console.error('Error fetching user details:', error);
            }
        }

        fetchUserDetails();
    }, []);

    return (
        <View style={styles.container}>
            <Text style={styles.title}>Profil</Text>
            <View style={styles.tile}>
                <View style={styles.row}>
                    <MaterialIcons style={styles.icon} name="card-account-details-outline" size={40} color="#333" />
                    <Text style={styles.text}>Nazwa</Text>
                </View>
                <Text style={styles.text}>{userDetails.name}</Text>
            </View>
            <View style={styles.tile}>
                <View style={styles.row}>
                    <MaterialIcons style={styles.icon} name="email" size={40} color="#333" />
                    <Text style={styles.text}>Email</Text>
                </View>
                <Text style={styles.text}>{userDetails.email}</Text>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flexGrow: 1,
        padding: 20,
        backgroundColor: 'transparent'
    },
    tile: {
        flexDirection: 'column',
        padding: 10,
        margin: 10,
        borderBottomWidth: 1,
        borderColor: COLORS.mainGrey,
    },
    row: {
        flexDirection: 'row',
        marginBottom: 10,
    },
    icon: {
        fontSize: 30,
        color: COLORS.third,
    },
    title: {
        fontSize: 20,
        // borderBottomWidth: 1,
        // paddingBottom: 20,
        marginBottom: 10,
        color: COLORS.third,
        // borderBottomColor: COLORS.third,
        textTransform: 'uppercase',
        fontWeight: 'bold',
        textAlign: 'left',
    },
    profileDetails: {
        flexDirection: 'row',
        alignItems: 'center',
        marginBottom: 20,
    },
    label: {
        fontWeight: 'bold',
        marginRight: 10,
        width: 100,
    },
    value: {
        flex: 1,
    },
    text: {
        marginLeft: 10,
        textAlign: "left",
        color: COLORS.third,
        fontSize: 22,
    }
});

export default ProfileScreen;
