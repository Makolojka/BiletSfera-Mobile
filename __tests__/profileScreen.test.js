import React from 'react';
import renderer from 'react-test-renderer';
import ProfileScreen from "../components/screens/ProfileScreen";

jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
test('renders correctly', () => {
    const tree = renderer.create(<ProfileScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});

