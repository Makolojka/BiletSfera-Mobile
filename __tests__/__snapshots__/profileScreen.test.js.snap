// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "flexGrow": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#2d1a42",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
        "textAlign": "left",
        "textTransform": "uppercase",
      }
    }
  >
    Profil
  </Text>
  <View
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#cccccc",
        "flexDirection": "column",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#333",
              "fontSize": 40,
            },
            {
              "color": "#2d1a42",
              "fontSize": 30,
            },
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰶫
      </Text>
      <Text
        style={
          {
            "color": "#2d1a42",
            "fontSize": 22,
            "marginLeft": 10,
            "textAlign": "left",
          }
        }
      >
        Nazwa
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#2d1a42",
          "fontSize": 22,
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    />
  </View>
  <View
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#cccccc",
        "flexDirection": "column",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#333",
              "fontSize": 40,
            },
            {
              "color": "#2d1a42",
              "fontSize": 30,
            },
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰇮
      </Text>
      <Text
        style={
          {
            "color": "#2d1a42",
            "fontSize": 22,
            "marginLeft": 10,
            "textAlign": "left",
          }
        }
      >
        Email
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#2d1a42",
          "fontSize": 22,
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    />
  </View>
</View>
`;
